backends/p4tools/modules/testgen/lib/test_backend.cpp

```diff
@@ -149,6 +149,7 @@ bool TestBackEnd::run(const FinalState &state) {
         if (testgenOptions.stopMetric == "MAX_NODE_COVERAGE" && coverage == 1.0) {
             return true;
         }
+        maxCoverage = std::max(maxCoverage, coverage);
```

Comment:
`coverage` is monotonically increasing, so this might not be necessary. We could just make `coverage` a class member.